### DISCUSSION TITLE
What are abstract classes for?

### DISCUSSION THREAD

**Discussion by haroldCoder** [community/community#76206] [Programming Help]
## class abstract Socket
![abstract1](https://github.com/community/community/assets/69125414/d49208af-9790-4d6f-afa0-f84e814f4b37)

## class UsersSocket
![abstract2](https://github.com/community/community/assets/69125414/0a86e28d-6fec-46c4-8020-38444eecfee3)

We often wonder about the usefulness of abstract classes. In this code snippet, an example of its application is clearly illustrated, especially when considering the use of sockets. Here, a Socket class is presented that establishes the socket connection and provides an abstract method to return data. The abstraction is evident by looking at the second class, which inherits from the first and returns a specific name.

The key lies in the abstract method, designed so that those who inherit from this class implement their own return (or cast) logic. Using this approach ensures that when you create an instance of the child class and call this method in another context, the resulting emit will be unique and suitable for the particular purpose of that instance.

#TypeScript #OOP

**Comment by ZhengLinLei** [comment]
Hi @haroldCoder,

You can check out this link on [Typescript abstract class](https://www.tutorialsteacher.com/typescript/abstract-class). Essentially, abstract classes are similar to interfaces, but you can pre-define properties and methods. By using the `abstract` keyword on methods, you are requiring your child classes to provide a definition for those methods.

And here you can find the answer of [Abstract classes vs interface](https://stackoverflow.com/a/50115567)
